src/NServiceBus.Metrics.AcceptanceTests/Tests/QueueLength/When_publishing_message.cs

```diff
@@ -38,26 +37,43 @@ public class When_publishing_message : NServiceBusAcceptanceTest
                     await session.Subscribe<TestEventMessage1>();
                     await session.Subscribe<TestEventMessage2>();
                 }))
-                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2)
+                .WithEndpoint<MonitoringSpy>()
+                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2 && SequencesReported(c))
                 .Run()
                 .ConfigureAwait(false);
+        }
 
-            var sessionIds = new[] { AssertHeaders(context.Headers1), AssertHeaders(context.Headers2) };
+        static bool SequencesReported(Context context)
```

Comment:
I do not think this is a good idea. SequencesReported will be executed many times by the done condition infrastructure and it contains business asserts that should be part of the test not burried in the done condition checking. I suggest to have a simple done condition and then do all the parsing and asserts after the await of the scenario in a dedicated Assert method.